src/github.com/matrix-org/dendrite/clientapi/sync/requestpool.go

```diff
@@ -95,7 +152,13 @@ func getTimeout(timeoutMS string) time.Duration {
 	return time.Duration(i) * time.Millisecond
 }
 
-// NewRequestPool makes a new RequestPool
-func NewRequestPool(db *storage.SyncServerDatabase) RequestPool {
-	return RequestPool{db}
+func getSyncStreamPosition(since string) (syncStreamPosition, error) {
+	if since == "" {
+		return syncStreamPosition(0), nil
```

Comment:
assuming the first event in the db gets an id of zero (does it?), and given `oldPos` is exclusive, I think this will mean the first event won't get returned?